strapi/strapi
# Fix relations for shortened contentTypes

### What does it do?

- Adds column to orderColumn attributes that are shortened already to prevent us from attempting to shorten again

### Why is it needed?

- Many-to-Many and Many-to-one relations are broken for contentTypes that need to be shortened 
- orderColumn is being shortened twice

### How to test it?

- create a collectionName with more than 55 characters, and try to create a many-to-many relation with another contentType or Many-to-one, it should work successfully with no errors
- you can follow the related issue for reproduction and confirming the fix 

### Related issue(s)/PR(s)

DX-1500

improves #20886 but there is still an issue with migrations causing the relational data to be lost
